osu.Game/Overlays/BeatmapSet/Buttons/PreviewButton.cs

```diff
@@ -61,9 +61,9 @@ public PreviewButton()
                     Origin = Anchor.Centre,
                     Size = new Vector2(18),
                 },
+                new HoverClickSounds()
```

Comment:
why go to the trouble of not inheriting `OsuClickableContainer` anymore if you're just going to jam `HoverClickSounds` back in, which is like half of the point of that class?

i swear this is all because of the baffling insistence on storing the button's enabled state in `playing.Disabled` instead of a separate `Enabled` bindable, which is both confusing and just doesn't fit the general design of everything around it (which is apparent by re-implementing button disable logic all over the place). i can't see a reasonable justification for it at all.

Reply:
no, it's literally because I can't find any way to do this cleanly without extra refactoring, `PlayButton` is disabling itself very innocently on some conditions, and with the current layout, `PreviewButton`, which is containing `PlayButton`, has to be disabled based on whether the play button is.

i've removed inheritance to `OsuClickableContainer` to remove exposing `Enabled` bindable and get rid of hiding properties and `base.Enabled` stuff to improve readability (and even get rid of public `Action` event while there, `OsuClickableContainer` isn't even correct for this type of button).

honestly at this point i'm just thinking about making `PreviewButton` inherit `PlayButton` and expose some method to create extra content around the button, it's the only thing that would make sense and would get this over with.